packages/react-ui/src/atoms/Typography.js

```diff
@@ -0,0 +1,31 @@
+import React, { forwardRef } from 'react';
+import PropTypes from 'prop-types';
+import { Typography as MuiTypography } from '@mui/material';
+
+const FontWeight = {
+  regular: 400,
+  medium: 500,
+  strong: 600
+};
+
+const Typography = forwardRef(({ italic, weight, children, ...otherProps }, ref) => {
+  // forwardRef needed to be able to hold a reference, in this way it can be a child for some Mui components, like Tooltip
+  // https://mui.com/material-ui/guides/composition/#caveat-with-refs
+
+  return (
+    <MuiTypography
+      {...otherProps}
+      ref={ref}
+      style={{ fontWeight: FontWeight[weight], fontStyle: italic && 'italic' }}
```

Comment:
Please include also `style` prop from user

In this shape, user is completly unable to overwrite `style` as we're ignoring /overwriting it.

Reply:
ok!

Reply:
Just putting the otherProps at the end would be enough, wouldn't it?

Reply:
> Just putting the otherProps 

@VictorVelarde 
It wouldn't work, 
 - `...otherProps` at end would overwrite this component style
 - style from `...otherProps` at beginning is overwritten in this version

Proper "merging" of properties "touched" by component must be manual, like this:

```jsx
function Outer({foo, style, className, ...otherProps}) 
{
    return <Inner 
       {...otherProps}
       className={clsx(foo && 'outerRoot', className)} 
       style={{color: 'red', ...style}}
    />
}
```

Reply:
Done, now styles overrides are working fine.

Reply:
Yeah, I know about override behaviour @zbigg ... my point here is Does it make sense to restrict _otherProps_ just to **style** prop and be explicit about it in the code? 

I was assuming here that if someone wants to play with _style_, beyond defaults, they can provide externally all they require (so no mix, but simple overriding), and allowing us to play with other props. Eg, check Typograghy options in https://mui.com/material-ui/api/typography/ and think about `gutterBottom`, `noWrap` or others.

Reply:
> I was assuming here that if someone wants to play with style, beyond defaults, they can provide externally all they require 

I would assume differently and would expect our "design system" components behave like proper MUI components. In this case, that "user provided style" and "props-based style" are properly merged _if_ they reasonably not collide.

For example with normal `Typography`, if you say `<Typography "color="primary" style={{cursor: "help"}}>` it just works and we don't care how typography "implements" color.

Nor, we're required to implement "color" by ourselves if we use "style". 

> provide externally all they require  

That would require user to know that `weight: "normal"` translates to `style={{fontWeight: 500}}` and that removes half of usefulness of this component.
(and workarounds for this would add even more api surface, like helper functions, exposing FontWeight mapping ...)

I think that robust APIs should ensure what unrelated features are indeed unrelated (as long as it's possible).